packages/record-data/addon/-private/ordered-set.ts

```diff
@@ -1,22 +1,118 @@
 import { assert } from '@ember/debug';
-import { guidFor } from '@ember/object/internals';
-import EmberOrderedSet from '@ember/ordered-set';
 
-export default class EmberDataOrderedSet<T> extends EmberOrderedSet<T> {
-  static create() {
-    return new this();
+type Dict<T> = import('@ember-data/store/-private/ts-interfaces/utils').Dict<T>;
+type RelationshipRecordData = import('./ts-interfaces/relationship-record-data').RelationshipRecordData;
+
+const NULL_POINTER = `null-${Date.now()}`;
+
+/*
+ TODO Ember's guidFor returns a new string per-object reference
+ while ours does not.
+
+ This has surfaced a bug during resurrection
+ in which Ember's guidFor would return false for `has` since the
+ resurrected record receives a fresh RecordData instance, leaving
+ the destroyed record in the set and thus depending on the state flags
+ for it not appearing elsewhere. We've accounted for this bug in the
+ updated OrderedSet implementation by doing a reference check: e.g.
+ the bug is preserved.
+
+ While we convert relationships to identifiers this will be something we
+ will be forced to address.
+*/
+function guidFor(obj): string {
+  if (obj === null) {
+    return NULL_POINTER;
+  }
+
+  return obj.clientId || obj.lid;
+}
```

Comment:
@snewcomer guidFor is the intended name for now. The `obj.lid` path is mostly here for when relationships store identifiers instead of RecordData instances. The concept of OrderedSet will go away soon so I don't think we need to rename this.